src/shims/unix/unnamed_socket.rs

```diff
@@ -347,13 +347,22 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         let this = self.eval_context_mut();
 
         let pipefd = this.deref_pointer_as(pipefd, this.machine.layouts.i32)?;
-        let flags = match flags {
+        let mut flags = match flags {
             Some(flags) => this.read_scalar(flags)?.to_i32()?,
             None => 0,
         };
 
         // As usual we ignore CLOEXEC.
         let cloexec = this.eval_libc_i32("O_CLOEXEC");
+
+        let mut is_nonblock = false;
+        // We need this check because `pipe` doesn't have any flags.
+        if flags != 0 {
```

Comment:
If `flags == 0` then `flags & this.eval_libc_i32("O_NONBLOCK") == this.eval_libc_i32("O_NONBLOCK")` will always be `false`, so -- why do we need this check?

Reply:
Oh yea, right :D